src/org/zaproxy/zap/extension/requester/ManualHttpRequestEditorPanel.java

```diff
@@ -0,0 +1,575 @@
+import org.zaproxy.zap.extension.httppanel.HttpPanel;
+import org.zaproxy.zap.extension.httppanel.HttpPanelRequest;
+import org.zaproxy.zap.extension.httppanel.HttpPanelResponse;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+
+public class ManualHttpRequestEditorPanel extends ManualRequestEditorPanel {
+
+	private static final long serialVersionUID = -5830450800029295419L;
+    private static final Logger logger = Logger.getLogger(ManualHttpRequestEditorPanel.class);
+
+	private ZapMenuItem menuItem;
+	
+	private HttpPanelSender sender;
+
+	private RequestResponsePanel requestResponsePanel;
+	private HttpPanelRequest requestPanel;
+	private HttpPanelResponse responsePanel;
+
+
+	private JToolBar footerToolbar = null;
+	// footer elements
+	private JLabel labelTimeElapse = null;
+	private JLabel labelContentLength = null;
+	private JLabel labelTotalLength = null;
+	private String helpKey = null;
+
+	public ManualHttpRequestEditorPanel(boolean isSendEnabled, String configurationKey) throws HeadlessException {
+		this(isSendEnabled, configurationKey, null);
+	}
+
+	public ManualHttpRequestEditorPanel(boolean isSendEnabled, String configurationKey, String helpKey) throws HeadlessException {
+		super(isSendEnabled, configurationKey);
+		this.helpKey = helpKey;
+		sender = new HttpPanelSender(getRequestPanel(), getResponsePanel());
+		
+		initialize();
+	}
+	
+	@Override
+	protected void initialize() {
+		super.initialize();
+
+		// add footer status bar
+		getWindowPanel().add(getFooterStatusBar(), BorderLayout.SOUTH);
+		
+		//setting footer status bar label and separator
+		getFooterStatusBar().add(getLabelTimeLapse());
+		getFooterStatusBar().addSeparator();
+		getFooterStatusBar().add(getLabelContentLength());
+		getFooterStatusBar().addSeparator();
+		getFooterStatusBar().add(getLabelTotalLength());
+	}
+	
+	@Override
+	public void setVisible(boolean show) {
+		super.setVisible(show);
+		
+		switchToTab(0);
+	}
+
+	@Override
+	public Class<? extends Message> getMessageType() {
+		return HttpMessage.class;
+	}
+
+	@Override
+	public Message getMessage() {
+		return getRequestPanel().getMessage();
+	}
+	
+	@Override
+	public void setMessage(Message aMessage) {
+		if (aMessage == null) {
+			return;
+		}
+		
+		getRequestPanel().setMessage(aMessage);
+		getResponsePanel().setMessage(aMessage);
+		setFooterStatus(null);
+		switchToTab(0);
+	}
+
+	@Override
+	protected MessageSender getMessageSender() {
+		return sender;
+	}
+	
+	@Override
+	protected HttpPanelRequest getRequestPanel() {
+		if (requestPanel == null) {
+			requestPanel = new HttpPanelRequest(true, configurationKey);
+			requestPanel.setEnableViewSelect(true);
+			requestPanel.loadConfig(Model.getSingleton().getOptionsParam().getConfig());
+		}
+		return requestPanel;
+	}
+	
+	private HttpPanelResponse getResponsePanel() {
+		if (responsePanel == null) {
+			responsePanel = new HttpPanelResponse(false, configurationKey);
```

Comment:
The HttpPanelRequest and HttpPanelResponse panels should be removed from HttpPanelManager once the ManualHttpRequestEditorPanel is no longer in use, otherwise this will lead to a memory leak.